src/lib.rs

```diff
@@ -27,3 +27,46 @@
 
 #[cfg(doc)]
```

Comment:
Should this file be renamed to something involving interning/contexts/whatever?

Reply:
I can move the context stuff into its own separate file, but the root file of a Rust crate has to be called `lib.rs`